app/components/Authentication/AwsEc2/AwsEc2.jsx

```diff
@@ -1,28 +1,704 @@
-import axios from 'axios';
-import React, { PropTypes } from 'react'
-import _ from 'lodash';
-import FlatButton from 'material-ui/FlatButton';
-import { green500, green400, red500, red300, yellow500, white } from 'material-ui/styles/colors.js'
-import { List, ListItem } from 'material-ui/List';
+import React, { PropTypes } from 'react';
+// Material UI
 import Dialog from 'material-ui/Dialog';
 import TextField from 'material-ui/TextField';
 import IconButton from 'material-ui/IconButton';
 import FontIcon from 'material-ui/FontIcon';
+import { Tabs, Tab } from 'material-ui/Tabs';
+import Paper from 'material-ui/Paper';
+import { List, ListItem } from 'material-ui/List';
+import FlatButton from 'material-ui/FlatButton';
+import { Toolbar, ToolbarGroup } from 'material-ui/Toolbar';
+import Subheader from 'material-ui/Subheader';
+import ActionAccountBox from 'material-ui/svg-icons/action/account-box';
+import ActionDelete from 'material-ui/svg-icons/action/delete';
+import ActionDeleteForever from 'material-ui/svg-icons/action/delete-forever';
+import Toggle from 'material-ui/Toggle';
+// Styles
+import styles from './awsec2.css';
+import sharedStyles from '../../shared/styles.css';
+import { green500, green400, red500, red300, yellow500, white } from 'material-ui/styles/colors.js';
+import Checkbox from 'material-ui/Checkbox';
+import { callVaultApi, tokenHasCapabilities } from '../../shared/VaultUtils.jsx';
+// Misc
+import _ from 'lodash';
+import update from 'immutability-helper';
+import Avatar from 'material-ui/Avatar';
+import { browserHistory } from 'react-router'
+import PolicyPicker from '../../shared/PolicyPicker/PolicyPicker.jsx'
+import VaultObjectDeleter from '../../shared/DeleteObject/DeleteObject.jsx'
+
+function snackBarMessage(message) {
+    let ev = new CustomEvent("snackbar", { detail: { message: message } });
+    document.dispatchEvent(ev);
+}
 
 export default class AwsEc2AuthBackend extends React.Component {
+
+    ec2ConfigSchema = {
```

Comment:
can we start placing this stuff in a config? Declaring undefined for everything doesn't seem right